### PR TITLE
Implement XUnit

### LINKED ISSUE #13: Discussion on unit testing framework
Since unit testing is good practice, it's probably best to start early before there's a lot of code in place.

Are there general preferences for unit test frameworks and ancillary libraries? I generally favor, and have had good results, with a combination of:

* XUnit
* NSubstitute
* AutoFixture
* Fluent Assertions

Those seem to be pretty common choices I see these days; the only thing I would note is that NSubstitute can be a little persnickety about mocking/stubbing some methods, so Rhino or Moq are always an option too.

I can take a cut at adding `test/` alongside the `src/` directory and setting up a basic unit test project, if desired.

### PR DESCRIPTION
**Pull request by edmistond**
## Description

Adds XUnit, AutoFixture, AutoFixture.XUnit2, FluentAssertions, and FluentAssertions.Analyzers. 

Moq or another mocking framework can be added once needed.

Also, corrected the namespace for the Webfinger endpoint to be Webfinger rather than Admin.

Fixes #13 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

Minor potentially breaking change since I moved the solution file up one level so it sits above the src and test directories, as well as correcting the references.

## How Has This Been Tested?

Builds clean, unit test passes.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Review by benrick — CHANGES_REQUESTED** [review]
Needs to adjust the GitHub Actions workflow. If I recall, it'll just need a different working directory.

**Comment by kirkbrauer** [comment]
@edmistond I can look at this as well, our entire company recently migrated to GitHub actions, we have a lot of .NET and Docker actions for all our DevOps.

**Comment by benrick** [comment]
> @edmistond I can look at this as well, our entire company recently migrated to GitHub actions, we have a lot of .NET and Docker actions for all our DevOps.

Yeah, I used to use Azure DevOps for work stuff (but wanted to use GitHub Actions) and then Microsoft acquired GitHub, and I could push that. I've used both. Both are good.

**Comment by edmistond** [comment]
@kirkbrauer It appears that yanking out the working dir fixed it 😄 

I've worked with Docker a fair bit, it's just that everywhere I've worked in the last 3-4 years or so used Azure DevOps. From looking at GH Actions, they seem a lot nicer and a slightly better thought out version of AzDo YAML build pipelines, which I've spent _far_ more time beating my head against than I really wanted to...

File: src/WebApp/Api/Webfinger/WebfingerEndpoints.cs [hunk 1]
```diff
@@ -1,6 +1,6 @@
 using Microsoft.AspNetCore.Mvc;
 
-namespace Smilodon.WebApp.Api.Admin;
```
Comment (benrick) [inline src/WebApp/Api/Webfinger/WebfingerEndpoints.cs:3]: Good catch. I missed this in my copy-pasta.

File: test/WebApp.Test/Api/Webfinger/WebfingerEndpointsTests.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+namespace WebApp.Test.Api.Webfinger
+{
+    public class WebfingerEndpointsTests
+    {
+        [Theory, AutoData]
+        public async Task GetAllAccountReturnsResource(string resourceName)
+        {
+            var result = await WebfingerEndpoints.GetAllAccount(resourceName);
+            result.Should().BeOfType<Ok<string>>()
+                .Which.Value.Should().Be(resourceName);
```
Comment (benrick) [inline test/WebApp.Test/Api/Webfinger/WebfingerEndpointsTests.cs:14]: 🤣 Glad to see my "hello world" works!

**Comment by kirkbrauer** [comment]
@edmistond Yeah, I found them to be a lot more bearable and there is so much open-source stuff to make your life easier on the GitHub Actions Marketplace. Really a no-brainer when starting a new project these days.

**Review by benrick — APPROVED** [review]
Looks like a good initial set up for this. 👍

File: test/WebApp.Test/Api/Webfinger/WebfingerEndpointsTests.cs (hunk 2 above)
Comment (edmistond) [reply test/WebApp.Test/Api/Webfinger/WebfingerEndpointsTests.cs:14]: It was either that or my usual "setting up a new unit test project and be sure they work" test of `true.Should().BeTrue();` 🤣

File: src/WebApp/Api/Webfinger/WebfingerEndpoints.cs [hunk 3]
```diff
@@ -1,6 +1,6 @@
...
+namespace Smilodon.WebApp.Api.Webfinger;
```
Comment (kirkbrauer) [inline src/WebApp/Api/Webfinger/WebfingerEndpoints.cs:3]: Should we try to make everything `WebFinger` for consistency?

Comment (benrick) [reply src/WebApp/Api/Webfinger/WebfingerEndpoints.cs:3]: Yes, you're right. That's how the spec has it. Want to send that as a separate PR?

Comment (kirkbrauer) [reply src/WebApp/Api/Webfinger/WebfingerEndpoints.cs:3]: Sure, can do. I also wanted to make all the property names consistent with the Microsoft Docs, which specify PascalCase for record properties, I can do a naming PR for that stuff.

**Pull request merged**